src/stratigraphy/groundwater/groundwater_extraction.py

```diff
@@ -0,0 +1,420 @@
+"""This module contains the GroundwaterLevelExtractor class."""
+
+import abc
+import logging
+from dataclasses import dataclass
+from datetime import datetime
+
+import fitz
+import numpy as np
+import regex
+from stratigraphy.groundwater.utility import extract_date, extract_depth, extract_elevation
+from stratigraphy.util.extract_text import extract_text_lines
+from stratigraphy.util.line import TextLine
+from stratigraphy.util.util import read_params
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class GroundwaterInformation(metaclass=abc.ABCMeta):
+    """Abstract class for Groundwater Information."""
+
+    depth: float  # Depth of the groundwater relative to the surface
+    date: str | None = (
+        None  # Date of the groundwater measurement, if several dates
+        # are present, the date of the document the last measurement is taken
+    )
```

Comment:
Why not already parse the extracted string and use a https://docs.python.org/3/library/datetime.html#datetime.date object here?

Reply:
Implemented in the latest version of the code.